src/v/serde/time_point.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "serde.h"
+
+#include <seastar/core/lowres_clock.hh>
+
+namespace serde {
+
+// Only system clocks time points are serializable. This is a loose check
+// but good enough, we assume any impls with to_time_t and from_time_t
+// are system clocks.
+template<class T>
+concept is_system_clock = requires {
+    T::clock::to_time_t;
+    T::clock::from_time_t;
+};
+
+// A serializable time_point type. Generally time points have a risk of
+// clock synchronization issues when deserialized on node with a clock drift.
+// Rather than letting users serialize arbitrary time_points, we only allow
+// serializing this special wrapped type making it easy for us to audit the
+// usage.
+// A time_point is serialized as the underlying duration epoch.
+template<is_system_clock time_point_type = ss::lowres_system_clock::time_point>
+struct time_point : envelope<time_point<time_point_type>, version<0>> {
+    using duration = typename time_point_type::duration;
+    using clock = typename time_point_type::clock;
+    time_point_type _time_point;
+
+    time_point() = default;
+
+    time_point(time_point_type t)
```

Comment:
I think I heard that this implicit conversion is intended, which probably deserves some documentation (and  some more thought on my part). If that is the case, does it make sense to have an `operator time_point_type() const`, too?

I guess some usage examples would help, perhaps serde roundtripping an `envelope` that contains a `time_point_type`?
```c++
struct foo : serde_envelope<foo, version<0>> {
   ss::lowres_system_clock::time_point tp;
};
```